src/opf/ed.jl

```diff
@@ -71,52 +67,35 @@ function build_opf(::Type{OPF}, network::Dict{String,Any}, optimizer;
 
     # Generation bounds (zero if generator is off)
     set_lower_bound.(pg, gen_status .* pgmin)
-    set_upper_bound.(pg, gen_status .* pgmax)
+    # ⚠️ we do not set upper bounds on pg because they are redundant
+    #       with the constraint pg + r <= pgmax
 
-    if OPF == EconomicDispatchWithReserves
-        # Reserve bounds (zero if generator is off)
-        set_lower_bound.(r, gen_status .* rmin)
-        set_upper_bound.(r, gen_status .* rmax)
-    end
+    # Reserve bounds (zero if generator is off)
+    set_lower_bound.(r, gen_status .* rmin)
+    set_upper_bound.(r, gen_status .* rmax)
 
     # Flow and flow slack bounds
-    for e in 1:E if !branch_status[e]
-            set_upper_bound(δf[e], 0.0)
-        end
-    end
     active_smax = branch_status .* smax
 
-    if soft_thermal_limit
-        # Soft thermal limits
-        # For active branches, sets pf + δf ≥ -smax and pf - δf ≤ smax and 0 ≤ δf ≤ Inf
-        # For inactive branches, sets pf + δf ≥ 0 and pf - δf ≤ 0 (pf = δf) and 0 ≤ δf ≤ 0 -> pf = 0
-        @constraint(model, pf_lower_bound[e in 1:E], pf[e] + δf[e] >= -active_smax[e])
-        @constraint(model, pf_upper_bound[e in 1:E], pf[e] - δf[e] <= active_smax[e])
-    else
-        # Hard thermal limits
-        # For active branches, sets pf ≥ -smax and pf ≤ smax
-        # For inactive branches, sets pf ≥ 0 and pf ≤ 0
-        @constraint(model, pf_lower_bound[e in 1:E], pf[e] >= -active_smax[e])
-        @constraint(model, pf_upper_bound[e in 1:E], pf[e] <= active_smax[e])
-    end
+    # Thermal limits
+    # Note that `soft_thermal_limit` is a boolean flag such that 
+    #   * if `true`, the penalty variables `δf` appears as expected
+    #   * if `false`, the penalty variables `δf` do not appear in the constraint
+    @constraint(model, pf_lower_bound[e in 1:E], pf[e] + soft_thermal_limit * δf[e] >= -active_smax[e])
+    @constraint(model, pf_upper_bound[e in 1:E], pf[e] - soft_thermal_limit * δf[e] <= active_smax[e])
```

Comment:
cc @klamike this is a non-trivial change to the code which we hadn't discussed before.
No change to functionality, but makes the code more compact